docs/testing/runtimes.md

```diff
@@ -0,0 +1,117 @@
+
+  - A task pane
+  - A function command
+  - A event-based (autolaunched) task
+  - A dialog (A dialog can be launched from either the task pane or the function command, but not from an event-based task. If a dialog is opened from both the task pane and the function command at the same time, then this add-in would have five runtimes.)
+
+## Share data across runtimes
+
+For Excel, PowerPoint, and Word add-ins, we recommend using a [Shared runtime](#shared-runtime) when any two or more features, except dialogs, need to share data. In Outlook, or scenarios where sharing a runtime isn't feasible, you need alternative methods. The parts of the add-in that are in separate runtime processes don't share global data automatically and are treated by the add-in's web application server as separate sessions, so [Window.sessionStorage](https://developer.mozilla.org/docs/Web/API/Window/sessionStorage) cannot be used to share data between them. The following are the recommended ways to share data *between unshared runtimes*.
+
+- Pass data between a dialog and its parent task pane, function command, or custom function by using the [Office.ui.messageParent](/javascript/api/office/office.ui#office-office-ui-messageparent-member(1)) and [Dialog.messageChild](/javascript/api/office/office.dialog#office-office-dialog-messagechild-member(1)) methods. 
+- To share data between a task pane and a function command, store data in [Window.localStorage](https://developer.mozilla.org/docs/Web/API/Window/localStorage), which is shared across all runtimes that access the same specific [origin](https://developer.mozilla.org/docs/Glossary/Origin). *LocalStorage isn't accessible in a JavaScript-only runtime and, thus, it isn't available in Excel custom functions or Outlook event-based tasks.*
+
+    > [!NOTE]
+    > Data in `Window.localStorage` persists between sessions of the add-in and is shared by add-ins with the same origin. Both of these characteristics are often undesirable for an add-in. You can ensure that each session of a given add-in starts fresh by calling the [Window.localStorage.clear](https://developer.mozilla.org/docs/Web/API/Storage/clear) method when the add-in starts. To allow some stored values to persist, but reinitialize other values, you can use [Window.localStorage.setItem](https://developer.mozilla.org/docs/Web/API/Storage/setItem) when the add-in starts for each item that should be reset to an initial value. You can also call [Window.localStorage.removeItem](https://developer.mozilla.org/docs/Web/API/Storage/removeItem) to delete an item entirely.
+
+- To share data between an Excel custom function and any other runtime, use [OfficeRuntime.storage](/javascript/api/office-runtime/officeruntime.storage).
+- To share data between an Outlook event-based task and any other runtime, use [OfficeRuntime.storage](/javascript/api/office-runtime/officeruntime.storage).
```

Comment:
OfficeRuntime.Storage works **only** for Event-Based Add-Ins in Outlook. To share data between Event-Based Add-Ins and other add-ins, such as taskpane, we recommend using the [SessionData API](https://docs.microsoft.com/en-us/javascript/api/outlook/office.sessiondata?view=outlook-js-preview).